### ISSUE TITLE
Enter key no longer submits the selection in raw mode

### ISSUE THREAD

**Issue by jkhsjdhjs** [flameshot-org/flameshot#1416]
**Flameshot version**

v0.9.0

**Describe the bug** 

The enter key no longer works for submitting a selection in raw mode (`flameshot gui -r`).

**To Reproduce**

Run `flameshot gui -r`, select an area and press the enter key. Nothing happens.

**Expected behavior**

Flameshot exits when pressing the enter key and prints the image to `stdout`.

**System Information**

Arch Linux, KDE Plasma on X11

**Comment by borgmanJeremy** [comment]
Thanks for reporting.  I think this was added with the new hotkey system. We will certainly fix it.

As a work around hit CTRL+S, and then close the save dialog.

**Comment by borgmanJeremy** [comment]
Issue was actuall introduced in this commit: 05c2bc6

**Comment by jfly** [comment]
@borgmanJeremy I'll investigate a bit further, but are you sure it was introduced by that commit? Enter is doing what I'd expect for me locally (I'm still running a patched version of 0.8.5, though). I'll rebase on top of the latest code and report back!

**Comment by borgmanJeremy** [comment]
Based on the results of my git bisect I think it was that commit.

**Comment by jfly** [comment]
Strange, I just did a build off of f54538e and this seems to work fine for me. Maybe there's something else going on here too?

**Comment by borgmanJeremy** [comment]
@jfly interesting.....what DE are you using?

**Comment by jfly** [comment]
I'm on xmonad

**Comment by borgmanJeremy** [comment]
Ok, here is what happens for me after that commit: 
![stuck](https://user-images.githubusercontent.com/46930769/109691770-6741dd00-7b4d-11eb-96a3-a13f337bf1fd.gif)

**Comment by jfly** [comment]
Oh woah. That looks different from what @jkhsjdhjs described. They said "nothing happens", but for you, the flameshot modal is clearly disappearing.

Wild-ass guess: is the flameshot process *actually* still running? It almost looks like it actually exited, and printed something to your terminal that broke your terminal...

**Comment by borgmanJeremy** [comment]
good call, here is what happened when I properly started the daemon. More closely matches the original issue
![stuck2](https://user-images.githubusercontent.com/46930769/109693033-cc4a0280-7b4e-11eb-9675-cbf28a6e7551.gif)

**Comment by borgmanJeremy** [comment]
I'll redo the bisect in a bit

**Comment by borgmanJeremy** [comment]
@jfly  Can you post your flameshot ini file?

**Comment by jfly** [comment]
Sure thing, here you go:

<details>
<summary>~/.config/flameshot/flameshot.ini</summary>

```
[General]
disabledTrayIcon=false
drawColor=#ff0000
drawThickness=3
ignoreUpdateToVersion=0.9.0
saveAfterCopyPath=/home/jeremy/Pictures
savePathFixed=false
showHelp=false
showStartupLaunchMessage=true
startupLaunch=false

[Shortcuts]
TYPE_ARROW=A
TYPE_CIRCLE=C
TYPE_CIRCLECOUNT=
TYPE_COMMIT_CURRENT_TOOL=Ctrl+Return
TYPE_COPY=Return
TYPE_DRAWER=D
TYPE_EXIT=Ctrl+Q
TYPE_IMAGEUPLOADER=
TYPE_MARKER=M
TYPE_MOVESELECTION=Ctrl+M
TYPE_MOVE_DOWN=J
TYPE_MOVE_LEFT=Left
TYPE_MOVE_RIGHT=Right
TYPE_MOVE_UP=Up
TYPE_OPEN_APP=Ctrl+O
TYPE_PENCIL=P
TYPE_PIN=
TYPE_PIXELATE=B
TYPE_RECTANGLE=R
TYPE_REDO=Ctrl+Shift+Z
TYPE_RESIZE_DOWN=Shift+Down
TYPE_RESIZE_LEFT=Shift+Left
TYPE_RESIZE_RIGHT=Shift+Right
TYPE_RESIZE_UP=Shift+Up
TYPE_SAVE=Ctrl+S
TYPE_SELECTION=S
TYPE_SELECTIONINDICATOR=
TYPE_SELECT_ALL=Ctrl+A
TYPE_TEXT=T
TYPE_TOGGLE_PANEL=Space
TYPE_UNDO=Ctrl+Z
```</details>

**Comment by borgmanJeremy** [comment]
Aha! Thats it. You have bound the enter key to 'copy'. The feature works if you hit copy or save, but the original behavior was such that return would always work. Thanks for helping even though this bug had nothing to do with your change :)

Now we need to decide on the correct fix. It could be a simple documentation update.

**Comment by FunWithUrban** [comment]
You mention it might only require a documentation update however I'm struggling to get this working the way I want. The shortcuts that were added in v0.9.0 looked to be exactly what I was waiting for in order to finally switch over to Flameshot. 

The workflow that I am trying to achieve is launching the program with a hotkey (which I see in the documentation is possible), select my screenshot region, then once ready upload the image to imgur and copy the the URL with a single shortcut key closing any Flameshot GUI elements in the process.

**Comment by borgmanJeremy** [comment]
@FunWithUrban I believe that all is possible (once i merge the fix for #1423. 

You would simply map "upload to imgur" to a hotkey. The issue here is related to the "raw" mode which has nothing to do with uploading to imgur.

**Comment by matidev** [comment]
> Aha! Thats it. You have bound the enter key to 'copy'. The feature works if you hit copy or save, but the original behavior was such that return would always work. Thanks for helping even though this bug had nothing to do with your change :)
> 
> Now we need to decide on the correct fix. It could be a simple documentation update.

- Change `TYPE_COPY=Return` that resolved that issue - it's my temporary fix :+1: 
- I installed Flameshot by Snap, so here is the `flameshot.ini` path: `../snap/flameshot/77/.config/flameshot`

**Comment by ottersarecool** [comment]
Just noticed that this isnt working anymore, will it ever make a comeback?

**Comment by FichteFoll** [comment]
Changing the hotkeys made it work just like before for me.

**Comment by jkhsjdhjs** [comment]
Yes, but I want the enter key to function without any additionally configuration in raw mode, just like it did before.
@borgmanJeremy said that it will certainly be fixed, but I fear it is not as certain anymore. I guess we have to await their decision here.

**Comment by mmahmoudian** [comment]
@jkhsjdhjs i wonder why are you insisting on having it work without configuration? Basically you configure it once and then for all the future versions it will stay the same.

**Comment by jkhsjdhjs** [comment]
The first reason is that it worked without configuration in the previous version. For me, it's a regression.
The second reason is, that in raw mode the image is printed to `stdout` on exit anyways, so there's no other location where you would want to store the image. Thus I don't see any problem with having the enter key bound to `print to stdout and exit` by default. This could also be a configurable option with `print to stdout and exit` as the default.

**Comment by mmahmoudian** [comment]
@jkhsjdhjs I'm sorry that you feel like this, but perhaps trying to see the bigger picture might help you understand the situation better. I just briefly touch upon some points that concerns your comment but I highly suggest following Flameshot as a software rather than a standalone tool you have on your computer:

1. The reason we now need configuration is that previously the keybindings were hardcoded, and now in the new version user has the freedom to change the keybinding to their preference

2. There is already a feature request in place (#1531) which concerns the raw mode you mentioned. Perhaps if you feel strongly about this, you should show your support in that thread instead


Of course I'm not the main dev, but the way I see it, Flameshot will never go back to hardcoded keybindings because it would be a destructive regression and a step backwards, and having a one-time config and sparing few kilobytes of the disk space for an ini file wouldn't hurt.

**Comment by jkhsjdhjs** [comment]
I understand that hardcoded keybindings are a bad thing and I agree the user should have the freedom to change these keybindings. So the configuration file is definitely a good thing. Most configuration files also have reasonable default values for options that aren't specified in the configuration file. That's why I propose a default value of `enter key` for submitting, perhaps only in raw mode if unwanted in default mode.

Although I fully support #1531 I don't think it would be the right place to discuss this as the other issue is about removing certain buttons from the UI while I request setting the default hotkey for `submit` in raw mode to the enter key. Also, this issue has been opened first. I agree that both issues are feature requests by now, but still requesting separate features.

**Comment by mmahmoudian** [comment]
In #1531 it is proposed to remove confusing buttons and add  a submit (or capture) button. When that button is implemented, it will have its keybinding which I agree that the default should be Enter
